nowcasting_utils/models/loss.py

```diff
@@ -20,20 +26,24 @@ def __init__(self, decay_rate: Optional[int] = None, forecast_length: int = 6):
         self.decay_rate = decay_rate
         self.forecast_length = forecast_length
 
-        logger.debug(f'Setting up weights with decay rate {decay_rate} and of length {forecast_length}')
+        logger.debug(
+            f"Setting up weights with decay rate {decay_rate} and of length {forecast_length}"
+        )
 
         # set default rate of ln(2) if not set
         if self.decay_rate is None:
             self.decay_rate = math.log(2)
 
         # make weights from decay rate
-        weights = torch.FloatTensor([math.exp(-self.decay_rate * i) for i in range(0, self.forecast_length)])
+        weights = torch.FloatTensor(
+            [math.exp(-self.decay_rate * i) for i in range(0, self.forecast_length)]
+        )
 
         # normalized the weights
         self.weights = weights / weights.sum()
 
         # move weights to gpu is needed
-        device = torch.device('cuda' if torch.cuda.is_available() else 'cpu')
+        device = torch.device("cuda" if torch.cuda.is_available() else "cpu")
         self.weights = self.weights.to(device)
```

Comment:
Does manually moving the weights to the GPU play nicely with Pytorch Lightning?  The [PyTorch Lightning docs](https://pytorch-lightning.readthedocs.io/en/stable/advanced/multi_gpu.html) say not to use `.to(device)` :)  (although I'm not sure what the alternative is here!)

Reply:
It does work okay, I noticed it fails when running the tests locally, where there is an available GPU, but the tests are running on the GPU. @peterdudfield and I chatted about it a bit, the other option seems to add a call to `.type_as(x)` in the computation, which would then run everytime its called, which might be slower, although we probably just need to profile it to check.

Reply:
Yeah, it might be worth trying `.type_as(x)` to see if it slows things down.  I'd hope the PyTorch folks have implemented `.type_as(x)` to be as quick as possible :)